### PR TITLE
internal authorization endpoint

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Ticket: [ALV-72](https://northerntech.atlassian.net/browse/ALV-72)

[ALV-72]: https://northerntech.atlassian.net/browse/ALV-72?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

File: app/app_iot_hub.go [hunk 1]
```diff
@@ -284,3 +288,51 @@ func (a *app) SetDeviceStateIoTHub(
...
+		if integration.Provider == model.ProviderIoTHub {
+			break
+		}
+	}
+	deviceModule := strings.SplitN(req.DeviceID, "/", 2)
+	for i := range deviceModule {
+		deviceModule[i] = url.PathEscape(deviceModule[i])
+	}
+	id := strings.Join(deviceModule, "/modules/")
+	log.FromContext(ctx).Warnf("getting twin: %s", id)
```
Comment (alfrunes) [inline app/app_iot_hub.go:308]:
```suggestion
```
This looks like a leftover debug log sent to `warn`.

File: app/app_iot_hub_test.go [hunk 2]
```diff
@@ -1284,3 +1285,110 @@ func TestSyncIoTHubDevices(t *testing.T) {
...
+
+		Integration model.Integration
+		Req         model.PreauthRequest
+
+		DataStore func(t *testing.T, self *testCase) *storeMocks.DataStore
+		Hub       func(t *testing.T, self *testCase) *hubMocks.Client
+
+		Error error
+	}
+	testCases := []testCase{{
```
Comment (alfrunes) [inline app/app_iot_hub_test.go:1317]: There are no test cases covering expected errors.

File: testdata/public.pem [hunk 3]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzogVU7RGDilbsoUt/DdH
+VJvcepl0A5+xzGQ50cq1VE/Dyyy8Zp0jzRXCnnu9nu395mAFSZGotZVr+sWEpO3c
+yC3VmXdBZmXmQdZqbdD/GuixJOYfqta2ytbIUPRXFN7/I7sgzxnXWBYXYmObYvdP
+okP0mQanY+WKxp7Q16pt1RoqoAd0kmV39g13rFl35muSHbSBoAW3GBF3gO+mF5Ty
+1ddp/XcgLOsmvNNjY+2HOD5F/RX0fs07mWnbD7x+xz7KEKjF+H7ZpkqCwmwCXaf0
+iyYyh1852rti3Afw4mDxuVSD7sd9ggvYMc0QHIpQNkD4YWOhNiE1AB0zH57VbUYG
+UwIDAQAB
+-----END PUBLIC KEY-----
```
Comment (alfrunes) [inline testdata/public.pem:9]: This could be inlined, unless you prefer otherwise.

**Review by alfrunes — CHANGES_REQUESTED** [review]
Just some leftover crap from my PoC...

File: app/app_iot_hub_test.go (hunk 2 above)
Comment (kjaskiewiczz) [reply app/app_iot_hub_test.go:1317]: yup; I was trying to cut some corners to speed things up, but I guess it doesn't make sense anymore, so I'll add more tests there

File: api/http/internal.go [hunk 4]
```diff
@@ -178,3 +179,44 @@ func (h *InternalHandler) BulkSetDeviceStatus(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, res)
 }
+
+// POST /tenants/:tenant_id/auth/{provider:[iot-hub]}
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply api/http/internal.go]: yeah, could you please squash this change to the parent commit?

File: api/http/devices.go.no [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+		}
+	}
+	if integration == nil {
+		rest.RenderError(c, http.StatusUnauthorized, errors.New("unauthorized"))
+		return
+	}
+	if strings.EqualFold(integration.Credentials.ConnectionString.HostName, hostName) {
+
+	}
+}
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply api/http/devices.go.no]: I squashed cleanup with your PoC commit

File: api/http/internal.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply api/http/internal.go]: done

File: app/app_iot_hub.go (hunk 1 above)
Comment (kjaskiewiczz) [reply app/app_iot_hub.go]: fixed

File: app/app_iot_hub_test.go (hunk 2 above)
Comment (kjaskiewiczz) [reply app/app_iot_hub_test.go]: fixed

File: testdata/public.pem (hunk 3 above)
Comment (kjaskiewiczz) [reply testdata/public.pem]: could be; done

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Pull request merged**